tests/core/gui/test_main_window.py

```diff
@@ -0,0 +1,194 @@
+# Parsec Cloud (https://parsec.cloud) Copyright (c) AGPLv3 2019 Scille SAS
+
+import pytest
+from PyQt5 import QtCore, QtWidgets
+
+from parsec.core.gui.lang import translate
+from parsec.core.types import BackendOrganizationFileLinkAddr
+
+
+@pytest.fixture
+async def logged_gui_with_files(
+    aqtbot, running_backend, backend, autoclose_dialog, logged_gui, bob, monkeypatch
+):
+    w_w = await logged_gui.test_switch_to_workspaces_widget()
+
+    assert logged_gui.tab_center.count() == 1
+
+    monkeypatch.setattr(
+        "parsec.core.gui.workspaces_widget.get_text_input", lambda *args, **kwargs: ("w1")
+    )
+    monkeypatch.setattr(
+        "parsec.core.gui.files_widget.get_text_input", lambda *args, **kwargs: ("dir1")
+    )
+
+    await aqtbot.mouse_click(w_w.button_add_workspace, QtCore.Qt.LeftButton)
+
+    def workspace_button_ready():
+        assert w_w.layout_workspaces.count() == 1
+        wk_button = w_w.layout_workspaces.itemAt(0).widget()
+        assert not isinstance(wk_button, QtWidgets.QLabel)
+
+    await aqtbot.wait_until(workspace_button_ready)
+
+    f_w = logged_gui.test_get_files_widget()
+    wk_button = w_w.layout_workspaces.itemAt(0).widget()
+    async with aqtbot.wait_exposed(f_w), aqtbot.wait_signal(f_w.folder_changed):
```

Comment:
we should not use `wait_signal` in the tests because it is cumbersome and buggy (most of the time we are waiting for the callback binded to the event to be executed, but we wait for the event in spite of a better thing to wait on)

```python
def _wait_files_widget_ready():
    assert f_w.isVisible()
    assert f_w.table_files.rowCount() == 4  # Wait for the list of files to display the correct number of items
await aqtbot.wait_until(_wait_files_widget_ready)
```

Reply:
As you notice it's done bellow, so in fact this check is pointless.